src/services/preference/types.ts

```diff
@@ -1,7 +1,12 @@
+import * as UserWhiteNoiseDataset from '../../datasets/user-white-noise.dataset';
+import * as DummyDataset from '../../datasets/dummy.dataset';
+
 export interface Preference {
   enabled: boolean;
 }
 
+export type PreferenceWithMeta<T extends Preference> = T & { _meta: PreferenceMeta<T> };
```

Comment:
Since almost all of the components only take and update current preference values and have nothing to do with the meta-information (default value, type, and other things like options for select type), separating them would be more convenient for us when developing components and directives to access `PreferenceService`.

For example, we can update the preference value without worrying about the existence of `_meta`:
https://github.com/googleinterns/guide-doge/blob/99a4025fe31cb404600e7d6c059043aab656f324/src/components/preference-group/preference-group.component.ts#L24

Reply:
Gotcha - sounds good.